src/main/java/ch/njol/skript/expressions/ExprFishingHook.java

```diff
@@ -0,0 +1,55 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ *
+ * Copyright 2011-2017 Peter Güttinger and contributors
+ */
+package ch.njol.skript.expressions;
+
+import org.bukkit.entity.FishHook;
+import org.bukkit.event.Event;
+import org.bukkit.event.entity.EntityDamageEvent.DamageCause;
+import org.bukkit.event.player.PlayerFishEvent.State;
+import org.eclipse.jdt.annotation.Nullable;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.expressions.base.EventValueExpression;
+import ch.njol.skript.lang.ExpressionType;
+
+@Name("Fishing Hook")
+@Description("The <a href='../classes.html#entity'>fishing hook</a> involved in a player fishing event.")
+@Examples({"on fishing:", "\tteleport player to fishing hook"})
+@Since("2.4")
+public class ExprFishingHook extends EventValueExpression<FishHook> {
```

Comment:
I don't really see the utility of this expression, is it just for documentation? You should be able to use `the fish hook` or `event-fish hook` since it's an entity.

Reply:
No, that's the point of the EventValueExpression is to be able to actually use it in that way.

Reply:
EventValueExpression is mainly (and I think only) used as default expression when registering a class and perhaps adding documentation to a specific event value but that's about it. You can use it this way since the [ExprEntity](https://github.com/SkriptLang/Skript/blob/master/src/main/java/ch/njol/skript/expressions/ExprEntity.java) handles it.

Being able to use `hook` instead of `fish hook` is something I'd rather avoid as well, in case it conflicts with something else in the future.

Reply:
'fishing' shouldn't be optional. Hook already means a few different things in Skript's source code.